src/equations/shallow_water_two_layer_2d.jl

```diff
@@ -0,0 +1,936 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+
+@doc raw"""
+    ShallowWaterTwoLayerEquations2D(gravity, H0, rho_upper, rho_lower)
+
+Two-Layer Shallow water equations (2LSWE) in two space dimension. The equations are given by
+```math
+\begin{alignat*}{8}
+&\frac{\partial}{\partial t}h_{upper}        
+&&+ \frac{\partial}{\partial x}\left(h_{upper} v_{1,upper}\right)
+&&+ \frac{\partial}{\partial y}\left(h_{upper} v_{2,upper}\right)  \quad 
+&&= \quad 0 \\
+&\frac{\partial}{\partial t}\left(h_{upper} v_{1,upper}\right)  
+&&+ \frac{\partial}{\partial x}\left(h_{upper} v_{1,upper}^2 + \frac{gh_{upper}^2}{2}\right) 
+&&+ \frac{\partial}{\partial y}\left(h_{upper} v_{1,upper} v_{2,upper}\right) \quad 
+&&= -gh_{upper}\frac{\partial}{\partial x}\left(b+h_{lower}\right) \\
+&\frac{\partial}{\partial t}\left(h_{upper} v_{2,upper}\right) 
+&&+ \frac{\partial}{\partial x}\left(h_{upper} v_{1,upper} v_{2,upper}\right) 
+&&+ \frac{\partial}{\partial y}\left(h_{upper} v_{2,upper}^2 + \frac{gh_{upper}^2}{2}\right) 
+&&= -gh_{upper}\frac{\partial}{\partial y}\left(b+h_{lower}\right)\\
+&\frac{\partial}{\partial t}h_{lower}  
+&&+ \frac{\partial}{\partial x}\left(h_{lower} v_{1,lower}\right) 
+&&+ \frac{\partial}{\partial y}\left(h_{lower} v_{2,lower}\right) 
+&&= \quad 0 \\
+&\frac{\partial}{\partial t}\left(h_{lower} v_{1,lower}\right) 
+&&+ \frac{\partial}{\partial x}\left(h_{lower} v_{1,lower}^2 + \frac{gh_{lower}^2}{2}\right) 
+&&+ \frac{\partial}{\partial y}\left(h_{lower} v_{1,lower} v_{2,lower}\right) 
+&&= -gh_{lower}\frac{\partial}{\partial x}\left(b+\frac{\rho_{upper}}{\rho_{lower}} h_{upper}\right)\\
+&\frac{\partial}{\partial t}\left(h_{lower} v_{2,lower}\right)  
+&&+ \frac{\partial}{\partial x}\left(h_{lower} v_{1,lower} v_{2,lower}\right) 
+&&+ \frac{\partial}{\partial y}\left(h_{lower} v_{2,lower}^2 + \frac{gh_{lower}^2}{2}\right) 
+&&= -gh_{lower}\frac{\partial}{\partial y}\left(b+\frac{\rho_{upper}}{\rho_{lower}} h_{upper}\right)
+\end{alignat*}
+```
+The unknown quantities of the 2LSWE are the water heights of the lower layer ``h_{lower}`` and the 
+upper 
+layer ``h_{upper}`` and the respective velocities in x-Direction ``v_{1,lower}`` and ``v_{1,upper}`` and in y-Direction
```

Comment:
```suggestion
layer ``h_{upper}`` and the respective velocities in x-direction ``v_{1,lower}`` and ``v_{1,upper}`` and in y-direction
```

Reply:
Done in 4f2114ca248344248107f39fd2d7e0523083b079